File: mobile-flutter/lib/models/follow.dart
/// Mirrors the web app's `Follow` (`web/src/types/follow.ts`) and the live
/// `follows` collection schema - `followerId`, `followingId`,
/// `followingName?`. `followingName` is denormalized at write time (there
/// is no `users` collection - see `plan/build-plan.md` "Known limitations")
/// so a profile screen can resolve a display name for a user who has never
/// posted. Same check-then-act uniqueness guard as [Like] - see
/// `FollowRepository.toggle`.
class Follow {
  const Follow({
    required this.id,
    required this.followerId,
    required this.followingId,
    required this.followingName,
  });

  factory Follow.fromJson(Map<String, dynamic> json) {
    return Follow(
      id: json['_id'] as String? ?? '',
      followerId: json['followerId'] as String? ?? '',
      followingId: json['followingId'] as String? ?? '',
      followingName: json['followingName'] as String?,
    );
  }

  final String id;
  final String followerId;
  final String followingId;
  final String? followingName;
}

File: go/internal/models/follow.go
package models

// Follow mirrors the `follows` Mudbase collection schema. FollowingName is denormalized at write
// time so a profile page can resolve a display name for a user who has never posted - there is no
// `users` collection in this schema (see store.ProfileService.ResolveDisplayName and README
// "Known limitations"). Same check-then-act uniqueness guard as Like.
type Follow struct {
	ID            string `json:"_id"`
	CreatedAt     string `json:"createdAt"`
	UpdatedAt     string `json:"updatedAt"`
	FollowerID    string `json:"followerId"`
	FollowingID   string `json:"followingId"`
	FollowingName string `json:"followingName,omitempty"`
}
